### ISSUE TITLE
Video to Audio ConvertProgressEvent not firing

### ISSUE THREAD

**Issue by amynkvirani** [AydinAdn/MediaToolkit#49]
When I try to convert a video .mpg to an audio .wav, both ConvertProgressEvent and ConversionCompleteEvent events are not fired.

**Comment by amynkvirani** [comment]
I found that someone else already fixed this and is available for you to merge. If you could do it, that would be great. Thanks

**Comment by tgiachi** [comment]
Please accept the Pull requests!

**Comment by vijaykishans** [comment]
Can somebody provide solution for this. I'm in urgent need of the same.
I believe regex could be an issue

**Comment by amynkvirani** [comment]
I believe it is already resolved with the last merge "699cb4a3dc14cfefe5ddaf1709f0de936c6533e0"

**Comment by vijaykishans** [comment]
Hi,

 

Thanks for your quick response. But Im unable to get the latest merge. How do I do that Pls help.

 

Regards

 

Vijay

**Comment by amynkvirani** [comment]
are you using the github code or used the nuget package?

**Comment by vijaykishans** [comment]
https://github.com/AydinAdn/MediaToolkit

 

Clicked on Clone or Download button

 

Added a console application. 

 

Used the below code as mentioned in readme but engine_ConversionCompleteEvent never gets triggered :

static void Main(string[] args)

        {

            StartConverting(); 

        }

      

        static public void StartConverting()

        {

            MediaFile objInputFiles = new MediaFile();

            var inputFile = new MediaFile { Filename = @"F:\ffmpeg\Baahubali2.mp4" };

            var outputFile = new MediaFile { Filename = @"F:\ffmpeg\encoded_Baahubali2.mp4" };

 

            using (var engine = new Engine())

            {

                engine.ConvertProgressEvent += ConvertProgressEvent;

                engine.ConversionCompleteEvent += engine_ConversionCompleteEvent;

                engine.Convert(inputFile, outputFile);

            }

        }

 

        static private void ConvertProgressEvent(object sender, ConvertProgressEventArgs e)

        {

            Console.WriteLine("\n------------\nConverting...\n------------");

            Console.WriteLine("Bitrate: {0}", e.Bitrate);

            Console.WriteLine("Fps: {0}", e.Fps);

            Console.WriteLine("Frame: {0}", e.Frame);

            Console.WriteLine("ProcessedDuration: {0}", e.ProcessedDuration);

            Console.WriteLine("SizeKb: {0}", e.SizeKb);

            Console.WriteLine("TotalDuration: {0}\n", e.TotalDuration);

            Console.ReadLine();

        }

 

      static  private void engine_ConversionCompleteEvent(object sender, ConversionCompleteEventArgs e)

        {

            Console.WriteLine("\n------------\nConversion complete!\n------------");

            Console.WriteLine("Bitrate: {0}", e.Bitrate);

            Console.WriteLine("Fps: {0}", e.Fps);

            Console.WriteLine("Frame: {0}", e.Frame);

            Console.WriteLine("ProcessedDuration: {0}", e.ProcessedDuration);

            Console.WriteLine("SizeKb: {0}", e.SizeKb);

            Console.WriteLine("TotalDuration: {0}\n", e.TotalDuration);

            Console.ReadLine();

        }

**Comment by vijaykishans** [comment]
Hi,
Am I missing something here?

regards
vijay

**Comment by amynkvirani** [comment]
remove the line `Console.ReadLine();` from function `engine_ConversionCompleteEvent` and `ConvertProgressEvent`

**Comment by vijaykishans** [comment]
Removed ReadLine stmts.

 

1.	Progress Event is working but not the engine_ConversionCompleteEvent.. On debug, it doesnot even get into the code.

 

Regards

 

Vijay

**Comment by amynkvirani** [comment]
After testing it myself, I was able to reproduce the same behavior i.e. ConversionCompleteEvent is never fired.
Unfortunately, after the last merge, this thing broke.
So what you can do is checkout the code at the following <a href="https://github.com/AydinAdn/MediaToolkit/tree/0fead8a8fcc745d07e3b88952e605a068c7e5e9d">commit</a> and try it out.

**Comment by vijaykishans** [comment]
Thanks a lot. I will try now and let you know my results.⁣

**Comment by vijaykishans** [comment]
Hi ,

 

Will I lose any important feature if I use this old commit ? How do I go about that. Please suggest.

 

Regards

 

Vijay

**Comment by amynkvirani** [comment]
I don't think that you will miss anything important. It was a stable version and should get your job done.

**Comment by dcosper** [comment]
The fix for this is as follows.
in RegexEngine.cs in the function
internal static bool IsProgressData

fps and frame are not sent in any of the examples i tried, so change your match requirements to
if (!matchSize.Success || !matchTime.Success ||
                 !matchBitrate.Success) return false;
As that is all we really need to monitor update progress
and if for some reason we get data with fps and frame change the below too

            long frame = 0;
            double fps = 0;
            if(matchFrame.Success)
                frame = Convert.ToInt64(matchFrame.Groups[1].Value, CultureInfo.InvariantCulture);
            if (matchFps.Success)
                fps = Convert.ToDouble(matchFps.Groups[1].Value, CultureInfo.InvariantCulture);

            int sizeKb = Convert.ToInt32(matchSize.Groups[1].Value, CultureInfo.InvariantCulture);
            double bitrate = Convert.ToDouble(matchBitrate.Groups[1].Value, CultureInfo.InvariantCulture);

this works for me to update progress
